crates/storage_impl/src/redis/cache.rs

```diff
@@ -98,41 +105,72 @@ pub enum CacheKind<'a> {
     All(Cow<'a, str>),
 }
 
+impl<'a> CacheKind<'a> {
+    pub fn from_redis_value(
+        value: RedisValue,
+    ) -> Result<Vec<Self>, Report<errors::ValidationError>> {
+        let value = value
+            .as_string()
+            .ok_or(errors::ValidationError::InvalidValue {
+                message: "Redis value not found in published message".into(),
```

Comment:
This should be more along the lines of not being able to convert the value to a string.